testing/acceptance_tests/circuit_test.py

```diff
@@ -0,0 +1,695 @@
+import pytest
+import os
+import sys
+import json
+import numpy as np
+import copy
+
+from zquantum.core import circuit
+from zquantum.core.utils import RNDSEED
+from zquantum.core.circuit import (
+    load_circuit_template_params,
+    save_circuit_template_params,
+    load_circuit,
+    save_circuit,
+    save_circuit_set,
+    load_circuit_set,
+    load_parameter_grid,
+    load_circuit_layers,
+    load_circuit_connectivity,
+    Circuit,
+)
+from zquantum.core.testing import create_random_circuit as _create_random_circuit
+
+sys.path.append("../..")
+from steps.circuit import (
+    build_ansatz_circuit,
+    create_random_circuit,
+    generate_random_ansatz_params,
+    combine_ansatz_params,
+    build_uniform_param_grid,
+    build_circuit_layers_and_connectivity,
+    add_ancilla_register_to_circuit,
+    concatenate_circuits,
+    batch_circuits,
+)
+
+
+class Test_generate_random_ansatz_params:
+    @pytest.mark.parametrize(
+        "number_of_layers, problem_size",
+        [
+            (1, 1),
+            (0, 1),
+            (1, 6),
+            (4, 3),
+        ],
+    )
+    def test_generate_random_ansatz_params_using_mock_ansatz_specs(
+        self, number_of_layers, problem_size
+    ):
+        # Given
+        ansatz_specs = {
+            "module_name": "zquantum.core.interfaces.mock_objects",
+            "function_name": "MockAnsatz",
+            "number_of_layers": number_of_layers,
+            "problem_size": problem_size,
+        }
+        seed = RNDSEED
+
+        filename = "params.json"
+        if os.path.exists(filename):
+            os.remove(filename)
+
+        # When
+        generate_random_ansatz_params(ansatz_specs=ansatz_specs, seed=seed)
+
+        # Then
+        assert os.path.exists(filename)
+        parameters = load_circuit_template_params(filename)
+        assert len(parameters) == number_of_layers
+        for parameter in parameters:
+            assert parameter < np.pi * 0.5
```

Comment:
I think these assertions should be removed.
If we ever change the default values of the parameters, these tests will start to fail.
If think it would be better to pass some `min_value` and `max_value` as input param and then check if parameters are set correctly.
And there could be a separate test which checks whether default values are `-pi/2;+pi/2`. WDYT?

Reply:
After thinking on it more, I think it just makes sense to remove the assertions that the parameters are within the given range. I feel like that's something we can _trust_ since it's really coming from numpy. Thoughts?

Reply:
Addressed here: 9e6a714